lib/reporters/base.js

```diff
@@ -211,6 +211,23 @@ var generateDiff = (exports.generateDiff = function (actual, expected) {
   }
 });
 
+/**
+ * This function dumps long stack traces for exceptions having
+ *  an Error cause() method
+ *
+ * @public
+ * @memberof Mocha.reporters.Base
+ * @param {Error} ex
+ * @return {string}
+ */
+exports.getFullErrorStack = function (err, message) {
+  var ret = err.stack || message || '';
+  if (err.cause) {
+    ret += '\n' + '   ' + 'Caused by: ' + exports.getFullErrorStack(err.cause);
```

Comment:
[Style] Just nitpicking, any reason for multiple strings?

```suggestion
    ret += '\n   ' + 'Caused by: ' + exports.getFullErrorStack(err.cause);
```

Reply:
wouldn't it be better to use same approach of `runner.js` and not do it recursively?

Reply:
generally for code reuse and performance. not that I think it is a bottleneck but both codes exist in the system already

Reply:
Fixed concatenation + reworked how the filtering was operating in `runner.js`, to make it focused on simply filtering.